File: test/qtsocket/runnable.cpp
#include <QRunnable>
#include <QByteArray>
#include <QTcpSocket>
#include <QThreadPool>

class Runnable : public QRunnable {
public:
    Runnable(qintptr socketDescriptor, ConnectionHandler *handler)
        : sockDesc(socketDescriptor), connectionHandler(handler) {}

    void run() override {
        qDebug() << "Runnable started";
        Comunicator *com = new Comunicator(sockDesc);
        if (!com->begin()) {
            qDebug() << "Couldn't begin Comunicator";
            return;
        }

        connect(com, &Comunicator::dataReceived, connectionHandler, &ConnectionHandler::onDataReceived);
        connect(com, &Comunicator::requestDelete, connectionHandler, &ConnectionHandler::removeMe);
    }

private:
    qintptr sockDesc;
    ConnectionHandler *connectionHandler;
};

File: test/qtsocket/runnable.py
from PyQt5.QtCore import QRunnable, QByteArray
from PyQt5.QtNetwork import QTcpSocket

from comunicator import Comunicator


class Runnable(QRunnable):
    
    canSafeDelete = False
    
    def __init__(self, sockd, cl):
        super().__init__()
        self.d = sockd
        self.cl = cl
        print("Runnable - initialized")
        
    def run(self):
        com = Comunicator(self.d)
        if not com.begin():
            print( "Coldnt Begin")
            self.canSafeDelete = True
            return
        com.dataReceived[QByteArray].connect(self.cl.onDataReceived)
        com.requestDelete.connect(self.cl.removeMe)
        self.cl.comunicators.append(com)